// Copyright 2016 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
////////////////////////////////////////////////////////////////////////////////
//
// bookstore.proto
// Test proto for transcoding
syntax = "proto3";
package google.api_manager.transcoding;
message Biography {
  int64 year_born = 1;
  int64 year_died = 2;
  string text = 3;
}
message AuthorInfo {
  string first_name = 1;
  string last_name = 2;
  Biography bio = 3;
}
message Book {
  string author = 1;
  string name = 2;
  string title = 3;
  AuthorInfo author_info = 4;
}
message Shelf {
  string name = 1;
  string theme = 2;
}
message ListShelvesResponse {
  repeated Shelf shelves = 1;
}
message CreateShelfRequest {
  Shelf shelf = 1;
}
message GetShelfRequest {
  int64 shelf = 1;
}
message DeleteShelfRequest {
  int64 shelf = 1;
}
message ListBooksRequest {
  int64 shelf = 1;
}
message CreateBookRequest {
  int64 shelf = 1;
  Book book = 2;
}
message GetBookRequest {
  int64 shelf = 1;
  int64 book = 2;
}
message DeleteBookRequest {
  int64 shelf = 1;
  int64 book = 2;
}
